include/daemons.h

```diff
@@ -31,6 +31,19 @@ typedef struct
     char email    [EMAILSZ];
 }   regmaildb_req;
 
+// Request header just to calculate offsets.
+typedef struct {
+    size_t cb;
+    int operation;
+} regmaildb_req_header;
+
+// Storage struct that fits all possible messages.
+typedef struct {
+    union {
+        regmaildb_req_header header;
+        regmaildb_req regmaildb;
+    } m;
+} regmaildb_req_storage;
```

Comment:
There might be a better way to do this:

```
typedef struct {
    size cb;
    int operation;
} regmaildb_req_header;

typedef struct {
    regmaildb_req_header; /* not sure if anonymous nested structs work this way */
    [request type-specific fields ... ]
} regmaildb_req;
```

But let's try that later.

Reply:
This doesn't work. I notice I don't need to give a name to `union`, so I've removed it. This lets us do `req_storage.header` etc.

Reply:
Why doesn't it work though?

Reply:
```c
typedef struct {
  regmaildb_req_header;
  int xxx;
} test_t;

void f() {
    test_t t;
    t.cb = 100;
}
```

```
../../include/daemons.h:120:3: warning: declaration does not declare anything [-Wmissing-declarations]
  regmaildb_req_header;
  ^~~~~~~~~~~~~~~~~~~~

regmaild.c:890:7: error: no member named 'cb' in 'test_t'
    t.cb = 100;
    ~ ^
```

Reply:
Ah, ok, so anonymous members don't work this way. Too bad.